### DISCUSSION TITLE
Implementing Dynamic Language Switching through a button - websocket call

### DISCUSSION THREAD

**Discussion by KutalVolkan** [deepgram/community#672] [General help]
Hello everyone,

I am currently using the following WebSocket connection:
```
async with websockets.connect(
            'wss://api.deepgram.com/v1/listen?encoding=linear16&sample_rate=16000&channels=1&model=nova-2&smart_format=true',
            extra_headers=extra_headers) as ws:
```
I aim to dynamically change the language by pressing a button. I thought the following approach would work:
```
async with websockets.connect(
            'wss://api.deepgram.com/v1/listen?encoding=linear16&sample_rate=16000&channels=1&model=nova-2&smart_format=true&language={source_language}', 
            extra_headers=extra_headers) as ws:
```
Unfortunately, this does not work. I am using the streaming feature of Deepgram. How should I configure the WebSocket call to dynamically change languages by pressing a button?

Language Parameter and Values:
[https://developers.deepgram.com/docs/language](url)
[https://developers.deepgram.com/docs/models-languages-overview](url)

SOLVED: It was a logic mistake

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.
